### DISCUSSION TITLE
How to query nested json fields in a collection?

### DISCUSSION THREAD

**Discussion by vaha-b** [art049/odmantic#475] [Q&A]
Hi,

In my FastAPI app, I am trying to query for all author publications by his authorId, but I didn't find any results on how to query nested fields in odmantic. And the documentation doesn't seem to mention anything about it.

![authors_screensch](https://github.com/art049/odmantic/assets/55129308/bfec7605-29b5-4db7-ac03-a9c41026c0a8)

**Comment by Oliversinn** [comment]
Maybe if you define a `BaseModel` as your variable type you could achieve it.

```python
class Author(BaseModel):
    authorId: str
    name: str


class Article(Model):
    ...
    authors: list[Author]

# query
Article.authors.name == "K. Hemminki"
```

I haven't test it myself, but I hope it works.

**Reply by vaha-b** [reply]
Although, it seems logical as in MongoDB shell, any attribute after `name` produced an error. I tried it before.

For example, I need `db.papers.find({"papers.authors.authorId": ”1715113"})`

(papers/articles)
